### ISSUE TITLE
Longitudinal submodel predictions

### ISSUE THREAD

**Issue by leRequinRomain** [drizopoulos/JMbayes2#100]
I am working with the jm() function from the jmbayes2 package and have encountered a situation where I need some clarification.

I have fitted a mixed-effects model using lmer() with random slopes and intercepts. I then used this model as the longitudinal submodel in jm(), paired with a dummy survival model. The results from this setup have shown good accuracy in grouped cross-validation with new ID groups that were not part of the training dataset.

My understanding is that when using jm(), the longitudinal predictions incorporate both fixed and random effects from the model. However, predictions for new patient data using lmer() are based solely on the fixed effects.

Is it valid to use jm() in this manner?

**Comment by drizopoulos** [comment]
Note that `jm()` works with mixed models fitted by function `lme()` (package **nlme**) or function `mixed_model()` (package **GLMMadaptive**), not `lmer()` (package **lme4**).

Via the `predict()` method for `jm` objects, you can obtain predictions using only the fixed effects (by setting `predict(..., type = "mean_subject")`) or predictions conditional on the random effects (by setting `predict(..., type = "subject_specific")`).

**Comment by leRequinRomain** [comment]
I used lmer() to check the underlying assumptions (i.e. normality of residuals and multicollinearity) with the performance package and then applied the equivalent formula in lme(). I made subject-specific predictions from the lme() longitudinal submodel using jm() from JMBayes2. This was not time series data, a dummy survival model was used, and the goal was to make subject-specific predictions for new grouped data where the subject was not in the training set. This allows both fixed and random effects in the predictions, unlike lmer() which only uses fixed effects.

With the lmer() predict function that uses fixed effects alone, there was no improvement in RMSE compared to using the mean of the response variable. However, subject-specific predictions resulted in about a 30% improvement in RMSE compared to the mean response value. I haven’t seen this method in the vignettes or research papers and am concerned there might be data leakage or some other issue causing overly optimistic predictions.
